management/server/updatechannel.go

```diff
@@ -170,3 +189,52 @@ func (p *PeersUpdateManager) HasChannel(peerID string) bool {
 
 	return ok
 }
+
+// handlePeerMessageUpdate checks if the update message for a peer is new and should be sent.
+func (p *PeersUpdateManager) handlePeerMessageUpdate(ctx context.Context, peerID string, update *UpdateMessage) bool {
+	p.channelsMux.Lock()
```

Comment:
we should avoid the write lock here. 

please move the serial map number comparison and last map into the write lock in the send update method.